### ISSUE TITLE
`foo.cmx: Permission denied` when running `dune exec` and `dune build` in parallel

### ISSUE THREAD

**Issue by copy** [ocaml/dune#5642]
## Expected Behavior

`dune exec` should work even if dune is running in the background in watch mode.

## Actual Behavior

`dune exec` fails:

```
File "init.ml", line 1:                   
Error: I/O error: .foo.eobjs/native/foo.cmx: Permission denied
```

## Reproduction

1. Run `dune build -w foo.exe` in one terminal
1. Make a change to one of the source files
1. Run `dune exec ./foo.exe` in another terminal

It's easier to reproduce if building foo.exe is slow.

## Specifications

- Version of `dune` (output of `dune --version`): 3.1.1
- Version of `ocaml` (output of `ocamlc --version`): 4.14.0
- Operating system (distribution and version): Linux

**Comment by jberdine** [comment]
Related: similar errors often occur when running `dune build @check` while dune is running in watch mode. Maybe it is not supposed to work, but it would definitely be an improvement it it could.

**Comment by emillon** [comment]
Yes, it's not supported to run several dune processes that share the same `_build` directory. See #236, #3340.

**Comment by emillon** [comment]
Duplicate of #236
